### ISSUE TITLE
No type argument is provided for AbstractContextManager, which causes pyright fail to infer type

### ISSUE THREAD

**Issue by ArchGuyWu** [ftd2xx/ftd2xx#77]
**Describe the bug**
Pyright cannot infer the type to FTD2XX
 
**To Reproduce**
```python
import ftd2xx
with ftd2xx.open() as dev:
```
1. install ftd2xx module
2. install vscode
3. write the code

**Expected behavior**
Pyright can infer the type to FTD2XX

**Logs**
If applicable, add logs to help explain your problem.

**Desktop (please complete the following information):**
 - OS: [e.g. Windows, Linux] Windows
 - OS Version: [e.g. Ubuntu 20.04, Windows 10] Windows 10
 - Python Version: [e.g. 3.10] 3.11.7
 - FTDI driver version: 2.12.36.4

**Additional context**
[related issue](https://github.com/microsoft/pyright/issues/7100#issuecomment-1909678734)

**Comment by snmishra** [comment]
Would the following diff work?
```
diff --git a/ftd2xx/ftd2xx.py b/ftd2xx/ftd2xx.py
index 30067ba..82958ce 100644
--- a/ftd2xx/ftd2xx.py
+++ b/ftd2xx/ftd2xx.py
@@ -340,7 +340,7 @@ else:
         return None


-class FTD2XX(AbstractContextManager):
+class FTD2XX(AbstractContextManager["FTD2XX"]):
     """Class for communicating with an FTDI device

     Use :any:`open` or :any:`openEx` to create an instance of this class.
```

**Comment by ArchGuyWu** [comment]
Yes.

**Comment by snmishra** [comment]
Fixed in 1.3.6
